modules/auth0/app/provider-auth0-client.tf

```diff
@@ -0,0 +1,91 @@
+#
+# Fetch the Auth0 tenant component deployment in some stack
+#
+variable "auth0_tenant_component_name" {
+  description = "The name of the component"
+  type        = string
+  default     = "auth0/tenant"
+}
+
+variable "auth0_tenant_environment_name" {
+  description = "The name of the environment where the Auth0 tenant component is deployed. Defaults to the environment of the current stack."
+  type        = string
+  default     = ""
+}
+
+variable "auth0_tenant_stage_name" {
+  description = "The name of the stage where the Auth0 tenant component is deployed. Defaults to the stage of the current stack."
+  type        = string
+  default     = ""
+}
+
+variable "auth0_tenant_tenant_name" {
```

Comment:
Could we avoid the double `tenant`?

Reply:
plus we use the same pattern frequently for remote state component context

```
<component-name>_tenant_name
<component-name>_stage_name
<component-name>_environment_name
```

But the component name here is `auth0_tenant`

Reply:
Understandable, it just confused me when I saw it but makes total sense and I agree.